### PR TITLE
Replace all ActiveIssue(851) with ConditionalFact

### LINKED ISSUE #1265: Create ConditionalFact for NegotiateStream tests requiring setup
These tests will not pass unless manual setup has been done:
```
NegotiateStream_Tcp_Tests.NegotiateStream_Tcp_With_ExplicitUserNameAndPassword
NegotiateStream_Tcp_Tests.NegotiateStream_Tcp_With_ExplicitUserNameAndPassword_With_Upn
NegotiateStream_Tcp_Tests.NegotiateStream_Tcp_With_ExplicitUserNameAndPassword_With_Spn
```

We previously used #851 as an ActiveIssue to suppress these, but we really need a ConditionalFact that can conditionally run these when appropriate setup has been done.  When creating this ConditionalFact, consider ways for the Console output and/or attributes on the test that describe what setup is required.

### LINKED ISSUE #1275: Eliminate NegotiateStreamTestConfiguration test helper
The NegotiateStream tests rely on a NegotiateStreamTestConfiguration, a test helper class that captures several test parameters necessary for the tests.  But it only redirects to TestProperties, duplicates TestProperties names and makes it look like there are 2 different ways to configure tests.

The goal of this issue is to make the tests that require these test parameters directly use their equivalent TestProperties and to eliminate the NegotiateStreamTestConfiguration class.  Where necessary, add [ConditionalFact] to make the tests run only when these parameters have been set.

### PR DESCRIPTION
**Pull request by roncain**
Prior to these changes, issue #851 and #1265 were used as ActiveIssue
to disable tests that required manual setup.  After these changes,
all these tests now use ConditionalFact to detect whether the
manual setup has been done and 851 and 1265 are no longer used as
ActiveIssue.

These changes also merge 2 different forms of TestProperties
for username and password into a single pair.  And it eliminates
a separate configuration class used by the negotiation tests
because it was redundant with TestProperties.

Also, ConditionalFacts that had associated TestProperties
(ex: SPN_Available and TestProperties.NegotiateTestSpn)
were cleaned up to expose getters for those TestProperties.
This eliminated tests needing to know which TestProperty
was associated with each ConditionalFact.

Fixes #1265
Fixes #1275

### CODE REVIEW

**Comment by roncain** [comment]
@dotnet-bot test outerloop Windows_NT
@dotnet-bot test outerloop selfhost Windows_NT

**Comment by roncain** [comment]
@dotnet-bot test outerloop Windows_NT
@dotnet-bot test outerloop selfhost Windows_NT

**Comment by roncain** [comment]
@dotnet-bot test all outerloop please

**Comment by roncain** [comment]
Reviewers: this PR refers to issue #851 because I initially thought this would go in first and PR #1261 would merge on top.  But given the complexity of that merge, I think it is better that #1261 go in first, and I will rebase on top of it and merge.  So in reviewing, please look only at the new usage of [ConditionalFact] for all the manual tests formerly deactivated with #851.  When I merge, I will retain changes made by @StephenBonikowsky

File: src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs [hunk 1]
```diff
@@ -192,5 +206,37 @@ public static bool NTLM_Available()
...
+        }
+
+        // Returns the explicit password if available
+        public static string GetExplicitPassword()
+        {
+            return ConditionalTestDetectors.GetExplicitPassword();
+        }
+
+        // Gets the UPN if available
+        public static string GetUPN()
```
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs]: Reviewers: one observation about the changes -- before these changes a test using ConditionalFact like UPN_Available had to just magically *know* which TestProperty to use for the value, which was confusing.  With these changes, the condition *and* value checked by the condition are both here in the base class.  Therefore a test testing base.UPN_Available() can just use base.GetUPN() to get the value associated with that condition.

**Comment by roncain** [comment]
@dotnet-bot test outerloop Windows_NT
@dotnet-bot test outerloop selfhost Windows_NT

File: src/System.Private.ServiceModel/tests/Scenarios/Binding/Tcp/NetTcpBindingTests.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
```
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Scenarios/Binding/Tcp/NetTcpBindingTests.cs]: Reviewers: FYI -- this is unrelated to my PR but ToF could not compile this file without this due to the new use of Console.WriteLine (in ToF-only conditions).  So I'm fixing it as part of this PR to allow testing.

**Comment by roncain** [comment]
@dotnet-bot test all outerloop please

File: src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs [hunk 3]
```diff
@@ -143,13 +143,27 @@ public static bool Explicit_Credentials_Available()
                                      ConditionalTestDetectors.AreExplicitCredentialsAvailable);
         }
 
+        // Returns 'true' if explicit credentials and domain are available to use.
```
Comment (StephenBonikowsky) [inline src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs]: Update comment with correct info.

File: src/System.Private.ServiceModel/tests/Common/Infrastructure/tests/TestPropertiesTest.cs [hunk 4]
```diff
@@ -18,13 +18,7 @@ public static void TestProperties_Property_Names_Are_Initialized()
         // This test will fail to compile if these names are not generated.
         // It is not necessary to test all the known properties, just to
         // test that property name code generation occurred.
-        Assert.NotNull(TestProperties.BridgeResourceFolder_PropertyName);
-        Assert.NotNull(TestProperties.BridgeHost_PropertyName);
-        Assert.NotNull(TestProperties.BridgePort_PropertyName);
-        Assert.NotNull(TestProperties.UseFiddlerUrl_PropertyName);
-        Assert.NotNull(TestProperties.NegotiateTestDomain_PropertyName);
-        Assert.NotNull(TestProperties.NegotiateTestUserName_PropertyName);
-        Assert.NotNull(TestProperties.NegotiateTestPassword_PropertyName);
```
Comment (StephenBonikowsky) [inline src/System.Private.ServiceModel/tests/Common/Infrastructure/tests/TestPropertiesTest.cs]: I assume this is clean-up of old bridge related code?

File: src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Negotiate/NegotiateStream_Http_Tests.cs [hunk 5]
```diff
@@ -24,55 +24,59 @@ public class NegotiateStream_Http_Tests : ConditionalWcfTest
...
         bool root_Certificate_Installed = Root_Certificate_Installed();
-        if (!windows_Authentication_Available || !root_Certificate_Installed)
+        bool ambient_Credentials_Available = Ambient_Credentials_Available(); 
+        if (!windows_Authentication_Available || !root_Certificate_Installed || !ambient_Credentials_Available)
         {
             Console.WriteLine("---- Test SKIPPED --------------");
             Console.WriteLine("Attempting to run the test in ToF, a ConditionalFact evaluated as FALSE.");
             Console.WriteLine("Windows_Authentication_Available evaluated as {0}", windows_Authentication_Available);    
             Console.WriteLine("Root_Certificate_Installed evaluated as {0}", root_Certificate_Installed);
+            Console.WriteLine("Ambient_Credentials_Available as {0}", ambient_Credentials_Available);
```
Comment (StephenBonikowsky) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Negotiate/NegotiateStream_Http_Tests.cs]: Add "evaluated" to the message for consistencies sake?

File: src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Negotiate/NegotiateStream_Http_Tests.cs [hunk 6]
```diff
@@ -111,28 +115,38 @@ public static void NegotiateStream_Http_AmbientCredentials()
...
+        if (!windows_Authentication_Available || 
+            !root_Certificate_Installed || 
+            !explicit_Credentials_Available || 
+            !domain_Available)
         {
             Console.WriteLine("---- Test SKIPPED --------------");
             Console.WriteLine("Attempting to run the test in ToF, a ConditionalFact evaluated as FALSE.");
             Console.WriteLine("Windows_Authentication_Available evaluated as {0}", windows_Authentication_Available);    
             Console.WriteLine("Root_Certificate_Installed evaluated as {0}", root_Certificate_Installed);
+            Console.WriteLine("Explicit_Credentials_Available as {0}", explicit_Credentials_Available);
```
Comment (StephenBonikowsky) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Negotiate/NegotiateStream_Http_Tests.cs]: "evaluated"

File: src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Negotiate/NegotiateStream_Http_Tests.cs [hunk 7]
```diff
@@ -175,29 +189,32 @@ public static void NegotiateStream_Http_With_ExplicitUserNameAndPassword()
...
+        bool spn_Available = SPN_Available();
+        if (!windows_Authentication_Available || 
+            !root_Certificate_Installed || 
+            !spn_Available)
         {
             Console.WriteLine("---- Test SKIPPED --------------");
             Console.WriteLine("Attempting to run the test in ToF, a ConditionalFact evaluated as FALSE.");
             Console.WriteLine("Windows_Authentication_Available evaluated as {0}", windows_Authentication_Available);    
             Console.WriteLine("Root_Certificate_Installed evaluated as {0}", root_Certificate_Installed);
+            Console.WriteLine("SPN_Available evaluated as {0}", spn_Available);
```
Comment (StephenBonikowsky) [inline src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Negotiate/NegotiateStream_Http_Tests.cs]: "evaluated"

**Comment by StephenBonikowsky** [comment]
All my comments are NITs.
LGTM

File: src/System.Private.ServiceModel/tests/Common/Infrastructure/tests/TestPropertiesTest.cs (hunk 4 above)
Comment (roncain) [inline src/System.Private.ServiceModel/tests/Common/Infrastructure/tests/TestPropertiesTest.cs]: Yes -- the NegotiateTestUserName and password no longer exist and needed to be removed, so I took the opportunity to prune out other obsolete entries.  This test was never intended to probe every TestProperty -- just probing one is enough to prove code-gen happened.

**Comment by roncain** [comment]
@dotnet-bot test all outerloop please

**Comment by roncain** [comment]
Merging based on having passed all OS OuterLoops and KonK and NonK.  I will respond to post-merge feedback.

**Pull request merged**